src/rpc/Counters.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RPC_COUNTERS_H
+#define RPC_COUNTERS_H
+
+#include <chrono>
+#include <cstdint>
+#include <string>
+#include <mutex>
+#include <boost/json.hpp>
+
+namespace RPC
+{
+
+class Counters
+{
+private:
+    struct MethodInfo
+    {
+        MethodInfo() = default;
+
+        std::uint32_t started{0};
+        std::uint32_t finished{0};
+        std::uint32_t errored{0};
+        std::uint32_t forwarded{0};
+        std::chrono::microseconds duration{0};
+    };
+
+    std::mutex mutex_;
```

Comment:
Can we do this with atomics and remove the mutex? Might have to prepopulate `methodInfo_`

Reply:
If we can't remove the mutex, we could probably at least change the counts to atomic variables, change the mutex to a shared_mutex, and thus reduce the time spent under a unique lock.